sdk-cpp/include/do_exceptions.h

```diff
@@ -1,21 +1,46 @@
 #ifndef _DELIVERY_OPTIMIZATION_DO_EXCEPTIONS_H
 #define _DELIVERY_OPTIMIZATION_DO_EXCEPTIONS_H
 
+#if (!DO_DISABLE_EXCEPTIONS)
+
 #include <exception>
 #include <stdint.h>
 #include <system_error>
 
-#if defined(DO_INTERFACE_COM)
-#include <winerror.h>   // FAILED macro
-#endif
+#endif // !DO_DISABLE_EXCEPTIONS
 
 namespace microsoft
 {
 namespace deliveryoptimization
 {
 
+// Error Macros
+#ifndef RETURN_HR_IF_FAILED
+#define RETURN_HR_IF_FAILED(hr)  {  \
+    int32_t __hr = (hr);            \
+    if(FAILED(__hr)) return __hr;   \
+}
+#endif
+
+#ifndef FACILITY_DELIVERY_OPTIMIZATION 
+#define FACILITY_DELIVERY_OPTIMIZATION   208
+#endif
+
+#define HRESULT_FROM_SYSTEM_ERROR(x) ((int32_t)(x) <= 0 ? ((int32_t)(x)) : ((int32_t) (((x) & 0x0000FFFF) | (FACILITY_DELIVERY_OPTIMIZATION << 16) | 0x80000000)))
+
+#ifndef FAILED
+#define FAILED(hr) (((int32_t)(hr)) < 0)
+#endif
+
+#ifndef SUCCEEDED
+#define SUCCEEDED(hr) (((int32_t)(hr)) >= 0)
+#endif
```

Comment:
Without exceptions, need a way to identify failing error codes. Which is a little awkward because we draw from a variety of error codes. (HRESULTS, std::error_code, our own error code enum, etc)

In the case of std::error_code, these values are defined as positive integers, and so error checking macro fail/succeeded wouldn't catch these failures appropriatley.

I introduced HRESULT_FROM_SYSTEM_ERROR as a result, and swapped out the facility code with our own, but I'm not sure if this is the best approach.